### PR TITLE
Introduce webui2 version of kiwi images

### PR DESCRIPTION
**Pull request by hellcp**

### CODE REVIEW

**Comment by coolo** [comment]
Hi, please consider https://github.com/coolo/open-build-service/commit/273a6a4a232f179ccc7a537f5489411218109941 and https://github.com/coolo/open-build-service/commit/202f7da806a70eb2091bdb03e7351e8f6346ff6e

**Comment by hellcp** [comment]
@coolo yep, linters are angry at me for that :D

**Comment by coolo** [comment]
While testing I noticed that the new jquery-datatables doesn't find its icons. I'm looking into it.

**Comment by coolo** [comment]
you don't need the datatables for kiwi images, no? As there is more to do with the datables switch, I'd cherry-pick into a different PR if that's fine with you

**Comment by hellcp** [comment]
Yeah, for sure

**Comment by coolo** [comment]
https://github.com/openSUSE/open-build-service/pull/5455

**Comment by hellcp** [comment]
I guess that layout needs some fixes so it displays breadcrumbs properly (although tests will still fail because in case of bootstrap dividers between breadcrumbs are a style elements)

**Comment by coolo** [comment]
you forgot one scss import

**Comment by coolo** [comment]
The test shouldn't be as specific about the style elements. Will have a look at it

**Comment by coolo** [comment]
There is no indication at all where we are:



Even the page title is just 'Open Build Service'. The whole personal navigation area does not appear very responsive - but this is a little beyond this PR.

So the only suggestion I have for you is this:
```
--- a/src/api/spec/features/webui/image_templates_spec.rb
+++ b/src/api/spec/features/webui/image_templates_spec.rb
@@ -74,6 +74,7 @@ RSpec.feature 'ImageTemplates', type: :feature, js: true do
 
       click_button('Create appliance')
       find('#kiwi-image-update-form')
+      return if ENV['BOOTSTRAP'] # TODO: no breadcrumbs
       expect(page).to have_text('home:tom:branches:my_project > package_with_kiwi_image')
     end
   end

**Comment by coolo** [comment]
Hmm, I just realize the test expects the 'Create Appliance' button to end up on the package page (which still has breadcrumbs)

**Comment by coolo** [comment]
scratch that - the breadcrumb the test expects is actually

"Open Build Service > Projects > home:tom:branches:my_project > package_with_kiwi_image > suse-13.2-live" and a good example why we should rather find a better solution than these runup lines

**Comment by coolo** [comment]
we could change it to expect a 'View Package' link pointing to home:tom:branches:my_project/package_with_kiwi_image

File: src/api/app/views/webui2/webui/kiwi/_tabs.html.haml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  $("#kiwi-image-details-section").hide();
+  });
+- kiwi_image_breadcrumb(@image, @image.name)
+.bg-light
+  %ul.nav.nav-tabs.pt-2.px-3
+    = content_tag('li', link_to('Overview', {controller: 'kiwi/images', action: 'show'}, class: "nav-link #{'active' if  action_name == 'show'}"), class: "nav-item" )
+    - link_to_details =  (action_name == 'edit') ? '#' : { controller: 'kiwi/images', action: 'edit', section: 'details' }
+    - link_to_software =  (action_name == 'edit') ? '#' : { controller: 'kiwi/images', action: 'edit', section: 'software' }
+    = content_tag('li', link_to('Details', link_to_details, { id: "kiwi_details_trigger", class: "nav-link #{'active' if @is_edit_details_action}"}), id: "kiwi_details_tab", class: "nav-item" )
+    = content_tag('li', link_to('Software', link_to_software, { id: "kiwi_software_trigger", class: "nav-link #{'active' if @is_edit_software_action}"}), id: "kiwi_software_tab", class: "nav-item" )
```
Comment (hennevogel) [inline src/api/app/views/webui2/webui/kiwi/_tabs.html.haml]: Also in general. The switch from `Details` to `Software` does not work. The switch from `Overview` to `Software` works

File: src/api/app/views/webui2/webui/kiwi/images/_repositories.html.haml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      This option will use the repositories from the current project. Other repositories set in this Kiwi Image will be REMOVED.
+%hr
+.card-body
+  #kiwi-repositories-list{ class: "#{'hidden' if f.object.use_project_repositories?}"}
+
+    %p#no-repositories{ class: "#{'hidden' if @image.repositories.present?}" }= 'There are no repositories.'
+    = f.fields_for :repositories do |repository_fields|
+      = render 'repository_fields', f: repository_fields
+    %p
+      = link_to_add_association(image_tag("drive_add", title: 'Add repository') + ' Add repository', f, :repositories)
```
Comment (hennevogel) [inline src/api/app/views/webui2/webui/kiwi/images/_repositories.html.haml]: This doesn't work anymore. It opened a modal before.

File: src/api/app/views/webui2/webui/kiwi/images/_packages.html.haml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.card-body
+  %h3 Packages
+  #kiwi-packages-list
+    %p#no-packages{ class: "#{'hidden' if @package_groups.packages.present?}" }= 'There are no packages.'
+
+    = f.fields_for :package_groups, @package_groups  do |package_group_fields|
+      = package_group_fields.fields_for :packages do |kiwi_package_fields|
+        = render 'package_fields', f: kiwi_package_fields
+      %p
+        = link_to_add_association(image_tag("package_add", title: 'Add package') + ' Add package', package_group_fields, :packages)
```
Comment (hennevogel) [inline src/api/app/views/webui2/webui/kiwi/images/_packages.html.haml]: This doesn't work anymore, it opened a model before.

File: src/api/app/views/webui2/webui/kiwi/images/_packages.html.haml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+.card-body
+  %h3 Packages
+  #kiwi-packages-list
+    %p#no-packages{ class: "#{'hidden' if @package_groups.packages.present?}" }= 'There are no packages.'
```
Comment (hennevogel) [inline src/api/app/views/webui2/webui/kiwi/images/_packages.html.haml]: Not sure why but I have packages and this message [in the review app](https://obs-reviewlab.opensuse.org/hellcp-webui2_kiwi/kiwi_images/1/edit?section=software#). 

screenshot from 2018-07-25 16-52-59

**Comment by hellcp** [comment]
@coolo that needs to be fixed with layout fixes, both ready_function and breadcrumbs aren't implemented there properly :D

File: src/api/app/views/webui2/webui/kiwi/images/_build_info.html.haml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+.bg-light
+  %ul.nav.nav-tabs.pt-2.px-3
+    %li.nav-item
+      %a.nav-link.active.result-select-link#result-select-link{ href: '#' } Build Results
+      = image_tag('icons/reload', title: 'Reload', onclick: 'update_build_result()', class: "result_reload", id: "result_reload")
```
[the remark this answers is not in the document]
Comment (hellcp) [reply src/api/app/views/webui2/webui/kiwi/images/_build_info.html.haml]: That's good to hear, I will fix that

File: src/api/app/views/webui2/webui/kiwi/images/_build_status.html.haml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+- @project = @image.package.project
+- @build_results.each_pair do |package, results|
+  #package-buildstatus
+    %table.table.table-stripped.table-sm.table-responsive
```
[the remark this answers is not in the document]
Comment (hellcp) [reply src/api/app/views/webui2/webui/kiwi/images/_build_status.html.haml]: Will use list group, thanks

File: src/api/app/views/webui2/webui/kiwi/images/_packages.html.haml (hunk 4 above)
Comment (hellcp) [reply src/api/app/views/webui2/webui/kiwi/images/_packages.html.haml]: Oh, forgot to replace hidden there :o

**Comment by coolo** [comment]
I won't miss breadcrumbs, but the ready_function needs to come back :)

**Comment by hennevogel** [comment]
[Bootstrap: Kiwi Images](https://trello.com/c/yE6Ms4H1/1496-bootstrap-kiwi-images)

File: src/api/app/views/webui2/webui/kiwi/_tabs.html.haml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+= content_for :ready_function do
```
[the remark this answers is not in the document]
Comment (bgeuken) [reply src/api/app/views/webui2/webui/kiwi/_tabs.html.haml]: I would prefer moving this into a javascript file. That way it's more visible which of our views need js.

File: src/api/app/views/webui2/webui/kiwi/_tabs.html.haml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  $("#kiwi_details_trigger").click(function() {
+  $("#kiwi-image-details-section").show();
```
Comment (bgeuken) [inline src/api/app/views/webui2/webui/kiwi/_tabs.html.haml]: Please indent the code inside of functions, or on newlines, by 2 characters:-)

**Comment by dmarcoux** [comment]
@hellcp Are you still working on this? We can help you moving this forward. Let us know if something is not clear. It would be a shame to leave it hanging.

**Comment by hellcp** [comment]
@dmarcoux yeah, very slowly ;P
Actually I encountered an error with docker image which prevents me from locally testing :/
```
Mysql2::Error: Table 'api_development.announcements' doesn't exist: SELECT `announcements`.* FROM `announcements` ORDER BY `announcements`.`created_at` DESC LIMIT 1 
```
seems that one of migrations didn't work out, can I actually run something to fix that?

**Comment by DavidKang** [comment]
@hellcp, you have a lot of offences in HAML views, because recently we enable it only for webui2. To check the offences have a look in our [WIKI](https://github.com/openSUSE/open-build-service/wiki/Development-Environment-Tips-&-Tricks#rake-tasks).

BTW, did you handle the issue that you had with the migrations?

**Comment by Conan-Kudo** [comment]
@hellcp Could you please rebase this so it can be integrated in?

**Comment by hellcp** [comment]
Moving to new and (hopefully) better PR

**Pull request closed without merging**